packages/admin/cms-admin/src/userPermissions/user/UserPageToolbar.tsx

```diff
@@ -44,24 +44,22 @@ export const UserPermissionsUserPageToolbar = ({ userId }: { userId: string }) =
             </ToolbarTitleItem>
             <FillSpace />
             <ToolbarActions>
-                {isAllowed("impersonation") && (
-                    <CrudMoreActionsMenu
-                        overallActions={[
-                            currentUser.impersonated
-                                ? {
-                                      icon: <Reset />,
-                                      label: commonImpersonationMessages.stopImpersonation,
-                                      onClick: () => stopImpersonation,
-                                  }
-                                : {
-                                      label: commonImpersonationMessages.startImpersonation,
-                                      icon: <ImpersonateUser />,
-                                      disabled: userId === currentUser.id,
-                                      onClick: () => startImpersonation(userId),
-                                  },
-                        ]}
-                    />
-                )}
+                <CrudMoreActionsMenu
+                    overallActions={[
+                        currentUser.impersonated
+                            ? {
+                                  icon: <Reset />,
+                                  label: commonImpersonationMessages.stopImpersonation,
+                                  onClick: () => stopImpersonation,
+                              }
+                            : {
+                                  label: commonImpersonationMessages.startImpersonation,
+                                  icon: <ImpersonateUser />,
+                                  disabled: !data.user.impersonationAllowed,
+                                  onClick: () => startImpersonation(userId),
+                              },
+                    ]}
+                />
```

Comment:
I think this check should stay since it's still theoretically possible that someone has only the `userPermissions` Permission but not the `impersonation` permission